Code/Engine/Foundation/Containers/Implementation/ArrayMap_inl.h

```diff
@@ -176,6 +176,18 @@ VALUE& ezArrayMapBase<KEY, VALUE>::GetValue(ezUInt32 index)
   return m_Data[index].value;
 }
 
+template <typename KEY, typename VALUE>
+EZ_ALWAYS_INLINE ezDynamicArray<typename ezArrayMapBase<KEY, VALUE>::Pair>& ezArrayMapBase<KEY, VALUE>::GetData()
```

Comment:
This would allow people to modify the keys and thus destroy the sorting. I would reset m_bSorted to false here, to guarantee a sorting step before the next lookup. This is not necessary for the const version.

Reply:
You could add a unit test for modifying the keys, and checking that the still lookup works afterwards, if you like, though I don't see that as super important.

Reply:
Good point! I was torn on exposing the method publicly or not doing it and making the begin and end methods friends, Marc favored the public method, so I did that. Should've then added at least the m_bSorted update for the non-const method and a test tho, totally right, I'll do that now.